### PR TITLE
Add support for exists queries and create OperationTypes

### PR DESCRIPTION
**Pull request by connernilsen**

### CODE REVIEW

File: service/src/test/java/com.codeforcommunity/processor/ProcessorImplTest.java [hunk 1]
```diff
@@ -19,4 +25,12 @@ void setup() {
...
     assertTrue(true);
   }
+
+  @Test
+  public void testFetchExists() {
+    DSLContext db = mockDb.getContext();
+    mockDb.addExistsReturn();
+    db.fetchExists(db.selectFrom(Tables.USERS).where(Tables.USERS.ID.eq(1)));
+    assertEquals(1, mockDb.getSqlOperationBindings().get(OperationType.EXISTS).size());
+  }
```
Comment (connernilsen) [inline service/src/test/java/com.codeforcommunity/processor/ProcessorImplTest.java:35]: Let me know if this test works for you. I'm getting an exception about reflection issues whenever I try to run `JooqMock` (even on that didn't happen before) on Java 8.

File: service/src/test/java/com.codeforcommunity/JooqMock.java [hunk 2]
```diff
@@ -198,11 +195,43 @@ int getCallCount() {
     }
   }
 
-  /** Constructor for JooqMock. 'UNKNOWN' and 'DROP/CREATE' operations are added by default. */
+  /**
+   * Represents an operation type in SQL. For all explicitly unsupported types you want to use, the
+   * {@link OperationType#UNKNOWN} type acts as a catchall.
+   */
+  public enum OperationType {
```
Comment (connernilsen) [inline service/src/test/java/com.codeforcommunity/JooqMock.java:202]: I created this enum and deprecated methods that used strings to determine the operation to perform.

**Review by liammoyn — COMMENTED** [review]
Shouldn't you be able to specify if you want the exists call to return true or false?

**Comment by connernilsen** [comment]
Oh lmao good point @liammoyn

File: service/src/test/java/com.codeforcommunity/processor/ProcessorImplTest.java (hunk 1 above)
Comment (connernilsen) [reply service/src/test/java/com.codeforcommunity/processor/ProcessorImplTest.java]: Will be fixed in another PR

**Review by liammoyn — APPROVED** [review]
Looks good to me, lets get our tests updated soon so we can remove these deprecated methods

**Pull request merged**